### PR TITLE
[stable/jenkins] Add support for openshift route in jenkins

### PR DESCRIPTION
**Pull request by hamza3202**
**Problem**
We need to automatically create route in openshift when jenkins is deployed

**Solution**
We need to add a new resource file to create route.

**What this PR does?**
Add option to create openshift route when needed. Configurable values for route include

- Master.route.enabled
- Master.route.annotations
- Master.route.labels
- Master.route.path

- [x] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md

### CODE REVIEW

**Comment by maorfr** [comment]
Isn't a Route crated automatically if you create an Ingress?

**Comment by hamza3202** [comment]
@maorfr the route does get created automatically but I am not able to find out how to enable TLS on the created route. Plus since eventually the route is the thing being used by openshift so would be better to be able to create it directly.

**Comment by hamza3202** [comment]
@maorfr Any updates on this?

**Comment by maorfr** [comment]
I completely understand. I plan to bring this up tomorrow in the [charts call](https://github.com/helm/community/blob/master/communication.md#meetings) as there are currently no charts with OpenShift resources, and I would like to understand how to move forward with this.

Thanks for contributing!

File: stable/jenkins/values.yaml [hunk 1]
```diff
@@ -361,6 +370,7 @@ NetworkPolicy:
 rbac:
   install: false
   serviceAccountName: default
+  serviceAccountAnnotations: {}
```
Comment (maorfr) [inline stable/jenkins/values.yaml]: i suggest we keep this PR around Route. there is another PR in progress that handles RBAC.

File: stable/jenkins/templates/jenkins-master-route.yaml [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    targetPort: http
+  tls:
+    insecureEdgeTerminationPolicy: Redirect
+    termination: edge
+  to:
+    kind: Service
+    name: {{ template "jenkins.fullname" . }}
+    weight: 100
+  wildcardPolicy: None
+{{- end }}
```
Comment (maorfr) [inline stable/jenkins/templates/jenkins-master-route.yaml]: missing new line?

File: stable/jenkins/README.md [hunk 3]
```diff
@@ -80,6 +80,11 @@ The following tables list the configurable parameters of the Jenkins chart and t
 | `Master.ingress.labels`           | Ingress labels                       | `{}`                                                                         |
 | `Master.ingress.path`             | Ingress path                         | Not set                                                                         |
 | `Master.ingress.tls`              | Ingress TLS configuration            | `[]`                                                                         |
+| `Master.route.enabled`          | Enables openshift route      | `false`                                                                         |
+| `Master.route.apiVersion`       | Route API version                  | `route.openshift.io/v1`                                                         |
```
Comment (maorfr) [inline stable/jenkins/README.md]: do we really need this option exposed?

File: stable/jenkins/Chart.yaml [hunk 4]
```diff
@@ -1,6 +1,6 @@
 name: jenkins
 home: https://jenkins.io/
-version: 0.37.2
+version: 0.37.3
```
Comment (maorfr) [inline stable/jenkins/Chart.yaml]: lets bump this to 0.38.0

**Review by maorfr — CHANGES_REQUESTED** [review]

File: stable/jenkins/values.yaml (hunk 1 above)
Comment (hamza3202) [reply stable/jenkins/values.yaml]: @maorfr I am removing this but this is definitely needed to add TLS support to routes so please get it available asap, Thanks!

File: stable/jenkins/Chart.yaml (hunk 4 above)
Comment (hamza3202) [reply stable/jenkins/Chart.yaml]: @maorfr I have changed the version to 0.40.0 as 0.39.0 is already released

**Comment by hamza3202** [comment]
@maorfr I have made the requested changes.

**Review by maorfr — APPROVED** [review]

**Comment by maorfr** [comment]
hey @hamza3202,

lets get this merged!
can you resolve conflicts?

**Comment by hamza3202** [comment]
@maorfr Done

File: stable/jenkins/templates/jenkins-master-route.yaml [hunk 5]
```diff
@@ -0,0 +1,31 @@
+{{- if .Values.master.route.enabled }}
+apiVersion: route.openshift.io/v1
+kind: Route
+metadata:
+{{- if .Values.master.route.labels }}
```
Comment (maorfr) [inline stable/jenkins/templates/jenkins-master-route.yaml]: Should this line be here? If a user does not define labels, there would be no labels at all?

Comment (hamza3202) [reply stable/jenkins/templates/jenkins-master-route.yaml]: Copy Paste issue, Fixed!

Comment (maorfr) [reply stable/jenkins/templates/jenkins-master-route.yaml]: Thanks!

**Comment by maorfr** [comment]
/lgtm

Thanks for contributing, and congrats! This is the first OpenShift resource in this repository :)

**Pull request merged**